Assets/VRTK/Scripts/Utilities/VRTK_SDKTransformModify.cs

```diff
@@ -0,0 +1,75 @@
+﻿// SDK Transform Modify|Utilities|90064
+namespace VRTK
+{
+    using UnityEngine;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    [Serializable]
+    public class VRTK_SDKTransformModifiers
+    {
+        [Tooltip("An optional SDK Setup to use to determine when to modify the transform.")]
+        public VRTK_SDKSetup sdkType;
+        [Tooltip("An optional SDK controller type to use to determine when to modify the transform.")]
+        public SDK_BaseController.ControllerType controllerType;
+        [Tooltip("The new position to change the transform to.")]
+        public Vector3 position = Vector3.zero;
+        [Tooltip("The new rotation to change the transform to.")]
+        public Vector3 rotation = Vector3.zero;
+        [Tooltip("The new scale to change the transform to.")]
+        public Vector3 scale = Vector3.one;
+    }
+
+    /// <summary>
+    /// The SDK Transform Modify can be used to change a transform orientation at runtime based on the currently used SDK or SDK controller.
+    /// </summary>
+    public class VRTK_SDKTransformModify : MonoBehaviour
+    {
+        [Tooltip("The target transform to modify on enable. If this is left blank then the transform the script is attached to will be used.")]
+        public Transform target;
+        [Tooltip("A collection of SDK Transform Modifiers to change the given target transform for each specified SDK.")]
+        public List<VRTK_SDKTransformModifiers> overrides = new List<VRTK_SDKTransformModifiers>();
+
+        protected VRTK_SDKManager sdkManager;
+
+        protected virtual void OnEnable()
+        {
+            target = (target != null ? target : transform);
+            sdkManager = VRTK_SDKManager.instance;
+            if (sdkManager != null)
+            {
+                sdkManager.LoadedSetupChanged += LoadedSetupChanged;
+            }
+        }
+
+        protected virtual void OnDisable()
+        {
+            if (sdkManager != null)
+            {
+                sdkManager.LoadedSetupChanged -= LoadedSetupChanged;
+            }
+        }
+
+        private void LoadedSetupChanged(VRTK_SDKManager sender, VRTK_SDKManager.LoadedSetupChangeEventArgs e)
+        {
+            //attempt to find by the overall SDK set up to start with
+            VRTK_SDKTransformModifiers selectedModifier = overrides.FirstOrDefault(item => item.sdkType == sdkManager.loadedSetup);
+
+            //If no sdk set up is found or it is null then try and find by the SDK controller
+            if (selectedModifier == null)
+            {
+                SDK_BaseController.ControllerType currentController = VRTK_DeviceFinder.GetCurrentControllerType();
+                selectedModifier = overrides.FirstOrDefault(item => item.controllerType == currentController);
+            }
+
+            //If a modifier is found then change the transform
+            if (selectedModifier != null && target != null)
```

Comment:
Remove the check `target != null` here and use it right at the start of this method to bail out early `if (target == null)`. This will save some unnecessary computation since we don't use the resulting modifier anyway in case of no target.